File: 1019-squares-of-a-sorted-array/squares-of-a-sorted-array.py
class Solution:
    def sortedSquares(self, nums: List[int]) -> List[int]:
        # #Approach1
        ans = [0 for val in nums]
        ans_pointer = len(ans) - 1

        front = 0
        back = len(nums) - 1

        while(front <= back):
            if(nums[front] ** 2 > nums[back] ** 2):
                ans[ans_pointer] = nums[front] ** 2
                ans_pointer -=1
                front += 1
            else:
                ans[ans_pointer] = nums[back] ** 2
                ans_pointer -= 1
                back -= 1
            
        return ans


                

File: 1019-squares-of-a-sorted-array/squares-of-a-sorted-array.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var sortedSquares = function(nums) {
    let i = 0
    let j = nums.length - 1
    let ans = new Array(nums.length)
    let k = ans.length - 1
    while(k>=0){
        if (nums[j] ** 2 > nums[i] **2){
            ans[k] = nums[j] ** 2 
            j--
            k--
        }else{
            ans[k] = nums[i] ** 2
            i++
            k--
        }
    }
    return ans
};